src/js/actions/tools.js

```diff
@@ -790,6 +808,35 @@ define(function (require, exports) {
             modifiers: {},
             fn: _vectorMaskHandler
         });
+        var DEBOUNCE_DELAY = 200;
+        _vectorMaskPolicyHandler = synchronization.debounce(function () {
+            var vectorMaskMode = toolStore.getVectorMode();
+
+            if (vectorMaskMode) {
+                var pointerPolicyID = toolStore.getVectorMaskPolicyID();
+         
+                if (pointerPolicyID) {
+                    return this.transfer(policy.removePointerPolicies, pointerPolicyID)
+                        .bind(this)
+                        .then(function () {
+                            var area = _getVectorMaskPolicyBounds.call(this),
+                                pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
+                                    OS.eventKind.LEFT_MOUSE_DOWN,
+                                    {},
+                                    area);
+                        
+                            return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
+                        })
+                        .then(function (policyID) {
+                            this.dispatch(events.tool.VECTOR_MASK_POLICY_CHANGE, policyID);
+                        });
+                }
+            } else {
+                return Promise.resolve();
+            }
+        }.bind(this), this, DEBOUNCE_DELAY, false);
```

Comment:
FYI I don't think you need this `.bind(this)` since the second argument to `debounce` does it for you.